Index: lib/scaffolding_extensions.rb
===================================================================
--- lib/scaffolding_extensions.rb	(revision 31)
+++ lib/scaffolding_extensions.rb	(working copy)
@@ -291,7 +291,7 @@
         else
           items = reflection.klass.find(:all, :order => reflection.klass.scaffold_select_order, :conditions=>reflection.options[:conditions], :include=>reflection.klass.scaffold_include)
           items.sort! {|x,y| x.scaffold_name <=> y.scaffold_name} if reflection.klass.scaffold_include
-          select(record, foreign_key, items.collect{|i| [i.scaffold_name, i.id]}, {:include_blank=>true})
+          select(record, foreign_key, items.collect{|i| [i.scaffold_name, i.id]}, {:include_blank=>true}) + javascript_tag("prepareselect('#{record}_#{foreign_key}')")
         end
       end
     end
