Index: lib/scaffolding_extensions.rb
===================================================================
--- lib/scaffolding_extensions.rb
+++ lib/scaffolding_extensions.rb
@@ -531,9 +531,22 @@
             to_boolean_select_tag(options)
           when :file
             to_input_field_tag("file", options)
+          when :autocomplete
+            to_auto_complete_tag(@method_name, options)
         end
       end
       
+      # Returns an auto completing text box not linked to an association.
+      # Useful when dealing with denormalized databases.  You need to define
+      # your own autocomplete action in the controller if you use this.
+      include ActionView::Helpers::JavaScriptHelper
+      include ActionView::Helpers::JavaScriptMacrosHelper
+      include ActionView::Helpers::UrlHelper
+      def to_auto_complete_tag(field, options)
+        klass = object.class.to_s.downcase
+        "#{object.class.scaffold_auto_complete_skip_style ? '' : auto_complete_stylesheet}<input id='#{klass}_#{field}' name='#{klass}[#{field}]' size='#{options[:size] || 30}' type='text' value='#{value(object)}' /><div class='auto_complete' id='#{klass}_#{field}_auto_complete'></div>#{javascript_tag("var #{klass}_#{field}_auto_completer = new Ajax.Autocompleter('#{klass}_#{field}', '#{options[:update] || "#{klass}_#{field}_auto_complete"}', 'auto_complete_for_#{klass}_#{field}', {})")}"
+      end
+      
       # Returns three valued select widget, for null, false, and true, with the appropriate
       # value selected
       def to_boolean_select_tag(options = {})
